airflow/providers/amazon/aws/operators/emr_add_steps.py

```diff
@@ -62,7 +63,7 @@ def __init__(
     ):
         if kwargs.get('xcom_push') is not None:
             raise AirflowException("'xcom_push' was deprecated, use 'do_xcom_push' instead")
-        if not (job_flow_id is None) ^ (job_flow_name is None):
+        if not exactly_one(job_flow_id is not None, job_flow_name is not None):
```

Comment:
We can't use this in providers -- otherwise it means the provider will only work with Airflow 2.3+, and while it's possible we can do that, this change isn't worth imposing that limit..

Reply:
yup true forgot about that thanks